src/bp/core/messaging/messaging-webhooks.ts

```diff
@@ -0,0 +1,130 @@
+import axios from 'axios'
+import { IO } from 'botpress/sdk'
+import { EventEngine } from 'core/events'
+import { TYPES } from 'core/types'
+import { inject, injectable, postConstruct } from 'inversify'
+import { AppLifecycle, AppLifecycleEvents } from 'lifecycle'
+import { ConversationService } from './conversation-service'
+import { MessageService } from './message-service'
+
+@injectable()
+export class MessagingWebhooks {
+  constructor(
+    @inject(TYPES.EventEngine) private eventEngine: EventEngine,
+    @inject(TYPES.ConversationService) private conversationService: ConversationService,
+    @inject(TYPES.MessageService) private messageService: MessageService
+  ) {}
+
+  @postConstruct()
+  async init() {
+    await AppLifecycle.waitFor(AppLifecycleEvents.CONFIGURATION_LOADED)
+
+    // TODO: register webhooks by http
+    const incomingWebhooks = [
+      {
+        id: 'mfkt837834her',
+        url: 'http://localhost:3020/hook',
+        // TODO: This could be a regex (but regex is kind of garbadge to write)
+        // This scope could be even further scoped for a bot: 'app.myBotId.userMessage'
+        scopes: ['app.userMessage']
+      }
+    ]
+    const outgoingWebhooks = [{ id: 'ef98484ufeb93', url: 'http://localhost:3020/hook', scopes: ['app.botMessage'] }]
+
+    // TODO: we need to include the api key when calling the webhooks
+    // TODO: why is the api key defined per workspace user??
+    // const http = axios.create({ headers: { 'x-bp-api-key': key } })
+
+    this.eventEngine.register({
+      name: 'webhook.outgoing',
+      description: 'Calls webhooks for outgoing events',
+      // 99 is just before the order of channels (100)
+      order: 99,
+      direction: 'outgoing',
+      handler: async (event: IO.OutgoingEvent, next) => {
+        const conversation = (await this.conversationService.forBot(event.botId).get(event.threadId!))!
+        if (!conversation) {
+          // channel is not on messaging
+          return next(undefined, false, true)
+        }
+
+        // TODO: no way to get an event's corresponding message. Should they share the same id? Or messageId property on event?
+        const message = await this.messageService.forBot(event.botId).create(conversation.id, event.payload)
```

Comment:
This is a problem with a pretty easy fix. The messaging service needs to supply its own message id for the event to take as an id when it's creating an event. We also need to automatically create an message for outgoing event (so we no longer call messages.create in channel because the message will already have been created).

This might break some things that we assuming event ids were numbers (since now they'll be uuids).